src/plugins/ipmi_intf.c

```diff
@@ -399,6 +409,21 @@ ipmi_intf_socket_connect(struct ipmi_intf * intf)
 			continue;
 		}
 
+		if (params->bind_intf != NULL) {
+			#ifdef SO_BINDTODEVICE
```

Comment:
first, please place the `#ifdef` at the leftmost column.

second, if `SO_BINDTODEVICE` will ever change from a macro to an `enum` or `const`, this check will break. It's better to go the conventional feature-checking way, alter `configure.ac` with some extra `AC_TRY_COMPILE` to check if the code using `SO_BINDTODEVICE` can be compiled on the platform, then `AC_DEFINE()` some `HAS_BINDTODEVICE` macro, and then rely on that macro here.

Additional benefit to that approach is that you can check that autoconf definition in `ipmitool.1.in` as well and don't list in the `man` the option that is not supported on the platform.

Reply:
Thanks for the guidance here, and apologies for the delay in getting back.

I added the suggested `AC_TRY_COMPILE` macro, and used that throughout the source files, but I haven't found a good way to do a conditional change of the man page.  I considered adding an output variable based on the same `AC_TRY_COMPILE` check, which would be empty or hold the option help. That doesn't seem to work for multi-line strings, and would move the option help string all the way into configure.ac, which seems odd.

I note that we have an existing similar case with the dual bridge support for the `open` interface, where we check for feature support with an AC_TRY_COMPILE at configure time, but we always include the related options in the man page.

Would it be OK to tackle the platform-specific man page request in another PR (or do you have some more advice on how I should tackle this here)?